controllers/update_pods.go

```diff
@@ -154,6 +156,11 @@ func (updatePods) reconcile(ctx context.Context, r *FoundationDBClusterReconcile
 	return deletePodsForUpdates(ctx, r, cluster, adminClient, updates, logger)
 }
 
+func shouldRequeueDueToTerminatingPod(pod *corev1.Pod, cluster *fdbtypes.FoundationDBCluster) bool {
```

Comment:
In general I would say that this method is too small to be factored out into something separate. My assumption is you only factored it out to have a unit test for this, if so I would prefer to move the `cluster.ProcessGroupIsBeingRemoved(processGroup.ProcessGroupID)` check into this method.

Reply:
I have moved this in the `shouldRequeueDueToTerminatingPod()` function although I didn't added any additional test cases for that as we already have test cases to validate `cluster.ProcessGroupIsBeingRemoved`